cdc/tests/utils/rawkv_data/gen_data.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pingcap/kvproto/pkg/kvrpcpb"
+	"github.com/pingcap/log"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/tikv/client-go/v2/rawkv"
+	"go.uber.org/zap"
+	"golang.org/x/exp/rand"
+	"golang.org/x/sync/errgroup"
+)
+
+func generateTestData(keyIndex int) (key, value0, value1 []byte) {
+	key = []byte(fmt.Sprintf("indexInfo_:_pf01_:_APD0101_:_%019d", keyIndex))
+	value0 = []byte(fmt.Sprintf("v0_%020d", keyIndex))
+	value1 = []byte(fmt.Sprintf("v1_%020d%020d", keyIndex, keyIndex))
+	return key, value0, value1
+}
+
+func batchGenerateData(keyIndex int, keyCnt int) (keys, values0, values1 [][]byte) {
+	keys = make([][]byte, 0, keyCnt)
+	values0 = make([][]byte, 0, keyCnt)
+	values1 = make([][]byte, 0, keyCnt)
+	for idx := keyIndex; idx < keyIndex+keyCnt; idx++ {
+		key, value0, value1 := generateTestData(idx)
+		keys = append(keys, key)
+		values0 = append(values0, value0)
+		values1 = append(values1, value1)
+	}
+	return keys, values0, values1
+}
+
+func NewPutCommand() *cobra.Command {
+	return &cobra.Command{
+		Use:          "put",
+		Short:        "Put rawkv data into tikv",
+		SilenceUsage: true,
+		Args:         cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runPointPut(cmd)
+		},
+	}
+}
+
+func NewDeleteCommand() *cobra.Command {
+	return &cobra.Command{
+		Use:          "delete",
+		Short:        "Delete rawkv data from tikv",
+		SilenceUsage: true,
+		Args:         cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runPointDelete(cmd)
+		},
+	}
+}
+
+func runPointDelete(cmd *cobra.Command) error {
+	cfg := &Config{}
+	err := cfg.ParseFromFlags(cmd.Flags())
+	if err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+	cli, err := rawkv.NewClientWithOpts(ctx, []string{cfg.SrcPD}, rawkv.WithAPIVersion(kvrpcpb.APIVersion_V2))
+	if err != nil {
+		return err
+	}
+	defer cli.Close()
+
+	eg := new(errgroup.Group)
+	for i := cfg.StartIndex; i < cfg.Count+cfg.StartIndex; i++ {
+		i1 := i
```

Comment:
Can use `i := i`. It's a idiom of Go.

Search `req := req` in https://go.dev/doc/effective_go.